db/generate.js

```diff
@@ -0,0 +1,191 @@
+  if (bookingsOnRoom === undefined) {
+    return true;
+  }
+
+  // Iterate all current bookings on room and then check new start and end date overlapping or not
+  for (let i = 0; i < bookingsOnRoom.length; i += 1) {
+    const booking = bookingsOnRoom[i];
+
+    // Not overlapping condition
+    // newCheckIn -- newCheckout  << booking.checkIn
+    // (new checkin and checkout time both should be earlier than current booking's checkin)
+    // booking.checkOut << newCheckIn -- newCheckOut
+    // (new checkin and checkout time both should be later than current booking's checkout)
+
+    // Below condition is checking it is overlapped -> then return false
+    if (!((moment(startDate) < moment(booking.check_in)
+    && moment(endDate) <= moment(booking.check_in))
+    || (moment(startDate) >= moment(booking.check_out)
+    && moment(endDate) > moment(booking.check_out)))) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function randomCheckInOutOnRoom(roomId) {
+  const room = rooms[`${roomId}`];
+
+  let startDate = moment(randomDate(moment().toDate(), moment().add(2, 'months').toDate())).startOf('day').toDate();
+  let endDate = moment(startDate).add(randomIntFromInterval(room.min_night, room.max_night), 'days').startOf('day').toDate();
+  let trial = 0;
+
+  while (!isNotOverlapWithOtherBookingDates(roomId, startDate, endDate)) {
+    trial += 1;
+    if (trial > 1) {
+      return null;
+    }
+    startDate = moment(randomDate(moment().toDate(), moment().add(2, 'months').toDate())).startOf('day').toDate();
+    endDate = moment(startDate).add(randomIntFromInterval(room.min_night, room.max_night), 'days').startOf('day').toDate();
+  }
+
+  return {
+    check_in: startDate,
+    check_out: endDate,
+  };
+}
+
+// Random Bookings
+function generateRandomBooking() {
+  const roomId = randomIntFromInterval(0, rooms.length - 1);
+  const room = rooms[roomId];
+
+  const randomCheckInOutDates = randomCheckInOutOnRoom(roomId);
+
+  if (randomCheckInOutDates === null) {
+    return null;
+  }
+
+  const booking = {
+    roomId,
+    email: faker.internet.email(),
+    guests: {
+      adults: randomIntFromInterval(1, room.max_guest.adults),
+      children: randomIntFromInterval(0, room.max_guest.children),
+      infants: randomIntFromInterval(0, room.max_guest.infants),
+    },
+    check_in: randomCheckInOutDates.check_in,
+    check_out: randomCheckInOutDates.check_out,
+    createdAt: moment(randomCheckInOutDates.check_in).subtract(randomIntFromInterval(0, 30), 'days').toDate(),
+  };
+  return booking;
+}
+
+
+function generateRandomBookings(num) {
+  let book;
+  for (let i = 0; i < num; i += 1) {
+    book = generateRandomBooking();
+    if (book !== null) {
+      bookings.push(book);
+      if (bookingsByRoom[`${book.roomId}`] === undefined) {
+        bookingsByRoom[`${book.roomId}`] = [];
+      }
+      bookingsByRoom[`${book.roomId}`].push(book);
+    }
+  }
+}
+
+
+generateRandomRooms(1);
+// console.log(rooms);
```

Comment:
you may consider deleting debugging functions

Reply:
okay! I will fix it! Thanks 👍